### ISSUE TITLE
`triton inst ...` support for updating tags

### ISSUE THREAD

**Issue by trentm** [joyent/node-triton#88]
CloudAPI endpoints:

    AddMachineTags
    ReplaceMachineTags
    ListMachineTags
    GetMachineTag
    DeleteMachineTag
    DeleteMachineTags     Delete all tags from the machine

Design 1:

    triton inst tag list INST   # list inst's tags
            # shortcut: triton inst tags INST
    triton inst tag get INST NAME   # get a single tag value
    triton inst tag add INST [NAME=VALUE ...]     # add (but not overwrite) tags
            # or call this 'set'
            # TODO: also allow adding from a JSON file (a la options for
            #       tags on 'triton create')
            # Q: Allow "triton inst tag set INST NAME VALUE"? I.e. no '=' in there.
    triton inst tag replace INST [NAME=VALUE ...] # add (with overwrite) tags
            # or 'replace' could instead be 'triton inst tag add -f ...' to force
            # XXX not clear to me if 'ReplaceMachineTags' means replace *all*
            #     tags with the new object. TODO: check
    triton inst tag delete INST [--all|NAME,...]   # delete all or specific tags

Alternative design:

    triton inst tag-list INST
    triton inst tag-get INST NAME
    triton inst tag-set INST [NAME=VALUE ...]
    triton inst tag-replace INST [NAME=VALUE ...]
    triton inst tag-delete INST NAME [NAME...]

I don't really like the "Alternative design". Arguably the only benefits are:
(a) the depth of 'triton inst tag get' is large (but meh); and (b) we *could*
have a shorter 'triton inst tag INST NAME VALUE' to set a tag. I'm not a fan. :)

FWIW, a comparison to gcloud (note that tags don't have values there):

    gcloud compute instances add-tags INST --tags TAG[,TAG,...]
    gcloud compute instances remove-tags INST --tags TAG[,TAG,...]
    gcloud compute instances remove-tags INST --all

GCE *does* have key/value pairs for metadata:

    gcloud compute instances add-metadata INST --metadata KEY=VALUE[,KEY=VALUE,...]
    gcloud compute instances add-metadata INST --metadata-from-file KEY=FILE

**Comment by trentm** [comment]
@arekinath Thoughts on the design?

**Comment by trentm** [comment]
From discussion with @arekinath:  ReplaceMachineTags is a full tags object replacement. AddMachineTags doesn't actually guard against overwrite -- which is the 'tag set' behaviour we want anyway.

So, this design:

```
triton inst tag list INST       # shortcut: triton inst tags INST
triton inst tag get INST NAME
triton inst tag set INST [NAME=VALUE ...|-f FILE]
triton inst tag replace INST [NAME=VALUE ...|-f FILE]
triton inst tag delete INST [--all|NAME,...]
```

Note: Big warnings in 'replace' help output that this is a full replacement of
the tags object.

**Comment by trentm** [comment]
alex suggested: 'triton inst tag replace-all ...' instead of "replace" to clarify the full replacement.  I like that better.

```
triton inst tag list INST       # shortcut: triton inst tags INST
triton inst tag get INST NAME
triton inst tag set INST [NAME=VALUE ...|-f FILE]
triton inst tag replace-all INST [NAME=VALUE ...|-f FILE]
triton inst tag delete INST [--all|NAME,...]
```

**Comment by trentm** [comment]
Demo:

```
$ triton inst tag ls $INST
{}
$ triton inst tag set $INST foo=bar pi=3.14 really=true
{
    "foo": "bar",
    "pi": 3.14,
    "really": true
}
$ triton inst tag get $INST foo
bar
$ triton inst tag get $INST foo -j      # JSON to differentiate num/bool/string
"bar"
$ triton inst tag get $INST really -j
true

$ cat tags.json
{"foo": "bling"}
$ cat tags.kv
key=value
beep=boop
$ triton inst tag set $INST -f tags.json -f tags.kv
{
    "foo": "bling",
    "pi": 3.14,
    "really": true,
    "key": "value",
    "beep": "boop"
}

$ triton inst tag rm $INST key really
Deleted instance ae40f7e8 tag key
Deleted instance ae40f7e8 tag really
$ triton inst tag ls $INST
{
    "foo": "bling",
    "pi": 3.14,
    "beep": "boop"
}
$ triton inst tag rm -a $INST
Deleted all instance ae40f7e8 tags
$ triton inst tag ls $INST
{}

$ triton inst tag set $INST foo=bar pi=3.14 really=true
{
    "foo": "bar",
    "pi": 3.14,
    "really": true
}
$ triton inst tag replace-all $INST -f tags.kv
{
    "key": "value",
    "beep": "boop"
}
```

**Comment by trentm** [comment]
This is in the `insttags-issue88` feature branch for now.

**Comment by trentm** [comment]
this is merged and published in triton 4.5.0 now
